File: src/java/model/Comment.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package model;

/**
 *
 * @author bim26
 */
public class Comment {
    private int comment_id;
    private int post_id;
    private int user_id;
    private String first_name;
    private String last_name;
    private String comment_text;

    public Comment(int comment_id, int post_id, int user_id, String first_name, String last_name, String comment_text) {
        this.comment_id = comment_id;
        this.post_id = post_id;
        this.user_id = user_id;
        this.first_name = first_name;
        this.last_name = last_name;
        this.comment_text = comment_text;
    }

    public Comment() {
    }
    

    public int getComment_id() {
        return comment_id;
    }

    public void setComment_id(int comment_id) {
        this.comment_id = comment_id;
    }

    public int getPost_id() {
        return post_id;
    }

    public void setPost_id(int post_id) {
        this.post_id = post_id;
    }

    public int getUser_id() {
        return user_id;
    }

    public void setUser_id(int user_id) {
        this.user_id = user_id;
    }

    public String getFirst_name() {
        return first_name;
    }

    public void setFirst_name(String first_name) {
        this.first_name = first_name;
    }

    public String getLast_name() {
        return last_name;
    }

    public void setLast_name(String last_name) {
        this.last_name = last_name;
    }

    public String getComment_text() {
        return comment_text;
    }

    public void setComment_text(String comment_text) {
        this.comment_text = comment_text;
    }
    
    
}

File: web/assets/js/comment.js
/* 
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/JSP_Servlet/JavaScript.js to edit this template
 */


document.addEventListener('DOMContentLoaded', function() {
    document.querySelectorAll('.post-method').forEach(function(form) {
        form.addEventListener('submit', function(e) {
            e.preventDefault();
            var postId = this.querySelector('input[name="post_id"]').value;
            var commentContent = this.querySelector('textarea[name="commentContent"]').value;

            fetch('commentServlet', {
                method: 'POST',
                headers: {
                    'Content-Type': 'application/x-www-form-urlencoded',
                },
                body: 'post_id=' + postId + '&commentContent=' + encodeURIComponent(commentContent)
            })
            .then(response => response.json())
            .then(data => {
                if (data.success) {
                    // Create new comment element
                    var commentHtml = `
                        <div class="comment mb-2" style="margin-left: 20px;">
                            <div class="comment-header">
                                <img src="assets/profile_avt/${data.profilePic}" class="img-fluid rounded-circle avatar me-2" style="width: 30px; height: 30px;">
                                <small><strong>${data.firstName} ${data.lastName}</strong></small>
                            </div>
                            <div class="comment-body">
                                <p style="margin-bottom: 0;">${data.commentText}</p>
                            </div>
                        </div>
                    `;

                    // Add new comment to the post
                    var postElement = this.closest('.post');
                    var commentsContainer = postElement.querySelector('.post-comments');
                    commentsContainer.insertAdjacentHTML('beforeend', commentHtml);

                    // Clear the comment input
                    this.querySelector('textarea[name="commentContent"]').value = '';
                } else {
                    console.error('Error:', data.error);
                }
            })
            .catch((error) => {
                console.error('Error:', error);
            });
        });
    });
});